### ISSUE TITLE
Use go:embed for embedding files instead of vfsgen

### ISSUE THREAD

**Issue by techknowlogick** [go-gitea/gitea#17352]
### Feature Description

Now that we have a minimum of go1.16 set, we can use go's embed functionality to embed templates/web assets (css/js), and translations.

**Comment by 6543** [comment]
example:
https://github.com/woodpecker-ci/woodpecker/blob/master/web/web.go#L10-L11

**Comment by lafriks** [comment]
only we will have to do some restucture of code as you can not embed files that are not in the same directory or in child directory where golang code is where you are embedding it.

**Comment by RahulSurwade08** [comment]
Hey @techknowlogick , I would love to contribute to this issue under Hacktoberfest. How can I get started?

**Comment by 6543** [comment]
the main concern is that if we ose std embeded the compressed files would be decompressed and compressed on each request that indicated gzip support ...

... @techknowlogick should we close this issue?

**Comment by wxiaoguang** [comment]
There could be a comment in code for why the vfsgen is chosen, to help future developers know the background. Otherwise months later the knowledge may be lost.

**Comment by jolheiser** [comment]
As well, this is still a valid issue. silverwind posted above a potential https://github.com/vearutop/statigz that allows serving embedded files via gzip

**Comment by lunny** [comment]
> precompress

Doesn't vfsgen have done that?

**Comment by silverwind** [comment]
Yes, but only gzip. We want Brotli as well for another ~30% size reduction over gzip.

**Comment by silverwind** [comment]
Are we fine with dropping support for non-bindata builds (e.g. always embed, don't support non-embedded assets). I fail to see a real use case for non-bindata builds and it just complicates the implementation a lot.

**Comment by delvh** [comment]
Assuming `make watch` will still work: I'm okay with it.
I noticed that when I compile with `TAGS="bindata …" make watch`,
the frontend doesn't work anymore and as I only get the raw HTML and some browser alerts, but no CSS or JS…

**Comment by silverwind** [comment]
Indeed, watch mode depends on file in the file system. One solution is to have go run a reverse proxy on the `assets` path in dev mode, which would proxy asset requests to `webpack-dev-server` or similar.

**Comment by silverwind** [comment]
Thought generally I'd like to avoid `webpack-dev-server` if possible and use vite which also has one built in, but that of course requires migrating to vite first.

**Comment by paralin** [comment]
Done in #22887

**Comment by lunny** [comment]
Closed by #34692
